### PR TITLE
Add `README.md` for C library

### LINKED ISSUE #75: Add `README.md` for C library

### PR DESCRIPTION
**Pull request by omdxp**
# Description

Resolves #75 

## Type of change

- [x] documentation

### CODE REVIEW

File: c/CMakeLists.txt [hunk 1]
```diff
@@ -18,13 +18,15 @@ add_executable(test test/test.c)
 
 # Create the example executable
 add_executable(example example/main.c)
+add_executable(example1 example/main.cpp)
```
Comment (ZibanPirate) [inline c/CMakeLists.txt:21]:
```suggestion
add_executable(example2 example/main.cpp)
```

File: c/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  - [C/C++](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools)
+  - [CMake](https://marketplace.visualstudio.com/items?itemName=twxs.cmake)
+  - [CMake Tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools)
+  - [conan-extension](https://marketplace.visualstudio.com/items?itemName=konicy.conan-extension)
+
+# Get Started
+
+- Install build dependencies:
+  ```sh
+  conan install . --output-folder=build --build=missing
```
Comment (ZibanPirate) [inline c/README.md:20]:
```suggestion
  cd c && conan install . --output-folder=build --build=missing
```

**Review by ZibanPirate — APPROVED** [review]
let's use [ctest](https://cmake.org/cmake/help/book/mastering-cmake/chapter/Testing%20With%20CMake%20and%20CTest.html#testing-using-ctest) for running tests, not in this PR but we should consider using it at some point

File: c/README.md (hunk 2 above)
Comment (omdxp) [reply c/README.md:20]: Since this `README.md` file was inside the `c/` folder, I thought I should not add the `cd` command but I think this is more clear

**Comment by omdxp** [comment]
> let's use [ctest](https://cmake.org/cmake/help/book/mastering-cmake/chapter/Testing%20With%20CMake%20and%20CTest.html#testing-using-ctest) for running tests, not in this PR but we should consider using it at some point

How about [Check](https://github.com/libcheck/check)?

**Pull request merged**